private/rules/javadoc.bzl

```diff
@@ -24,15 +25,27 @@ def generate_javadoc(
     args.add("--element-list", element_list)
     args.add_all(source_jars, before_each = "--in")
     inputs.extend(source_jars)
+    args.add_all(ctx.files.doc_resources, before_each = "--resources")
+    inputs.extend(ctx.files.doc_resources)
     args.add_all(classpath, before_each = "--cp")
     transitive_inputs.append(classpath)
+
     for dep in doc_deps:
         dep_info = dep[_JavadocInfo]
         args.add("-linkoffline")
         args.add(dep_info.url)
         args.add(dep_info.element_list.dirname)
         inputs.append(dep_info.element_list)
-    args.add_all(javadocopts)
+
+    processed_opts = []
+    for opt in javadocopts:
+        if "$(" in opt:
```

Comment:
what's this doing?

Reply:
Is there a canonical example this was copied from to learn?

Reply:
We have a variable `maven_version` set in bazelrc and need to substitute that into the `--doctitle`. So inside the rule we have to resolve that variable.
It is similar to what is done for maven_coordinates https://github.com/bazelbuild/rules_jvm_external/blob/edbbe1d68bf156251f9f75d48387ca2d34a833e5/private/rules/maven_publish.bzl#L36